daemon/src/lib.rs

```diff
@@ -392,14 +394,17 @@ where
 
     #[instrument(skip(self), err)]
     pub async fn propose_settlement(&self, order_id: OrderId) -> Result<()> {
-        // TODO: we need to know which contract symbol the order is for
-        let latest_quote = self
+        let contract_symbol = self
+            .executor
+            .query(order_id, |cfd| Ok(cfd.contract_symbol()))
+            .await?;
+
+        let latest_quote = *self
             .price_feed_actor
-            .send(xtra_bitmex_price_feed::LatestQuote(
-                xtra_bitmex_price_feed::ContractSymbol::BtcUsd,
-            ))
+            .send(xtra_bitmex_price_feed::GetLatestQuotes)
```

Comment:
🔧 Since you've introduced the `GetLatestQuotes` message, we could just pass that map through to the `collab_settlement` protocol and use it here:

https://github.com/itchysats/itchysats/blob/a7b51027fe2b955280e3275321017d6dec41587a/daemon/src/collab_settlement/taker.rs#L53-L59

You'd have access to the CFD's contract symbol in the closure and would be able to pick the correct quote from the map before calling `start_collab_settlement_taker`.

This would save us from introducing the `query` API on `executor`, which is a bit suspicious.

Reply:
thanks for the suggestion, I'll give it a shot!

Reply:
it seems it can be done, but will move around a few things (we used to check invariants before we proposed settlement, now we'll need to check them *inside* taker_cfd, after it receives the proposal. Funnily enough, inside that actor we have access to the DB so I will be able to just fish out the contract symbol then from the hashmap. 

The awkward thing is that the `ProposeSettlement` will get a HashMap of latest quotes, instead of a concrete *proposal*, which is a particular bid and ask price at a given time.

I'll show you how it looks like and we can decide then. in the end, I don't think it will be cleaner :/

Reply:
I had a look too before proposing this and I thought it was weird to check invariants in a `TakerActorSystem` method anyway.

> The awkward thing is that the ProposeSettlement will get a HashMap of latest quotes, instead of a concrete proposal, which is a particular bid and ask price at a given time.

I thought we could move the feed to the `taker_cfd::Actor`. It's not used for anything else in the `TakerActorSystem`. That way you wouldn't have to include the `HashMap` in the message.